polars/polars-lazy/src/dsl/function_expr/sign.rs

```diff
@@ -0,0 +1,37 @@
+use super::*;
+use polars_core::export::num;
+use DataType::*;
+
+pub(super) fn sign(s: &Series) -> Result<Series> {
+    match s.dtype() {
+        Float32 => {
+            let ca = s.f32().unwrap();
+            sign_float(ca)
+        }
+        Float64 => {
+            let ca = s.f64().unwrap();
+            sign_float(ca)
+        }
+        _ => {
```

Comment:
we can match on `dt => if dt.is_numeric()` to only cast integer types.

Reply:
Done!